Version 0.13.0 broken on platforms where `libc::suseconds_t` != `i64`

On 32-bit platforms, `libc::suseconds_t` is usually `i32` instead of `i64`. This results in a compilation error like the one below, since `u32` cannot be converted into `i32` without the risk of wrapping.

```rust
 error[E0277]: the trait bound `i32: From<u32>` is not satisfied
  --> /cargo/registry/src/index.crates.io-6f17d22bba15001f/if-addrs-0.13.0/src/posix_not_mac.rs:75:50
   |
75 |                 tv_usec: timeout.subsec_micros().into(),
   |                                                  ^^^^ the trait `From<u32>` is not implemented for `i32`, which is required by `u32: Into<_>`
   |
```

--- a/src/posix_not_mac.rs
+++ b/src/posix_not_mac.rs
@@ -72,7 +72,10 @@ impl PosixIfChangeNotifier {
         let timeout = if let Some(timeout) = timeout {
             let mut t = timeval {
                 tv_sec: timeout.as_secs().try_into().expect("timeout overflow"),
-                tv_usec: timeout.subsec_micros().try_into().expect("timeout overflow"),
+                tv_usec: timeout
+                    .subsec_micros()
+                    .try_into()
+                    .expect("timeout overflow"),
             };
             // a timeout of 0 is infinity, so if the requested duration is too
             // small, make it nonzero

--- a/src/posix_not_mac.rs
+++ b/src/posix_not_mac.rs
@@ -72,7 +72,7 @@ impl PosixIfChangeNotifier {
         let timeout = if let Some(timeout) = timeout {
             let mut t = timeval {
                 tv_sec: timeout.as_secs().try_into().expect("timeout overflow"),
-                tv_usec: timeout.subsec_micros().into(),
+                tv_usec: timeout.subsec_micros().try_into().expect("timeout overflow"),
             };
             // a timeout of 0 is infinity, so if the requested duration is too
             // small, make it nonzero

